From e1aa165dfecde07ea117e8b233ea99c5590b7adb Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Fri, 23 Jul 2010 10:15:40 +0800
Subject: [PATCH 121/280] ENGR00125360 ipu lib: add debug level enviroment
 support

For better debug, adding debug level enviroment support for ipu lib.
Just export IPU_DBG=debug_level to set it.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 ipu/mxc_ipu_hl_lib.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/ipu/mxc_ipu_hl_lib.c b/ipu/mxc_ipu_hl_lib.c
index 7b8fd23..31e8998 100644
--- a/ipu/mxc_ipu_hl_lib.c
+++ b/ipu/mxc_ipu_hl_lib.c
@@ -2447,6 +2447,7 @@ int mxc_ipu_lib_task_init(ipu_lib_input_param_t * input,
 {
 	int ret = 0;
 	ipu_lib_priv_handle_t * ipu_priv_handle;
+	char * dbg_env;
 
 	dbg(DBG_INFO, "\033[0;34m*** mxc_ipu_lib_task_init ***\033[0m\n");
 
@@ -2456,6 +2457,12 @@ int mxc_ipu_lib_task_init(ipu_lib_input_param_t * input,
 		goto err0;
 	}
 
+	dbg_env = getenv("IPU_DBG");
+	if (dbg_env) {
+		debug_level = atoi(dbg_env);
+		dbg(DBG_INFO, "ipu debug level %d\n", debug_level);
+	}
+
 	if (!g_ipu_shm) {
 		if (_ipu_sem_prepare() < 0) {
 			ret = -1;
-- 
1.8.0

